src/Adapter/Translations/TranslationRouteFinder.php

```diff
@@ -204,6 +215,10 @@ private function isModuleUsingNewTranslationSystem($moduleName)
     {
         $module = $this->moduleRepository->getInstanceByName($moduleName);
 
+        if (!($module instanceof Module)) {
+            throw new Exception($this->translator->trans('Invalid module : %s', array($moduleName), 'Admin.International.Notification'));
```

Comment:
I think it would be best to throw an untranslated `PrestaShopBundle\Exception\InvalidModuleException`, which would have to be catched in `TranslationsController::modifyTranslationsAction()` and transformed into a proper error message.

For this you'd need to [add a flash message](https://symfony.com/doc/3.4/controller.html#flash-messages) and redirect back to the `admin_international_translations_show_settings` route.

And then, I think a better error message would be: "An error has occurred, this module does not exist: %s" which already already exists in `Admin.International.Notification`

Reply:
Thank you @eternoendless  ! Done